src/kde_analysis/eval_m1m2chieff_poiss_weights.py

```diff
@@ -206,79 +489,164 @@ def get_rate_m_chieff2D(m1_query, m2_query, Rate):
     m1 = samples[:, 0]  # First column corresponds to m1
     m2 = samples[:, 1]  # Second column corresponds to m2
     cf = samples[:, 2]
-    samples2 = np.vstack((m2, m1, cf)).T
-    # Combine both samples into one array
-    symmetric_samples = np.vstack((samples, samples2))
+    ########################
 
     # Determine weights based on vt_weights flag
     if boots_weighted:
         if vt_weights:
             weights_over_VT = poisson_weights / vt_vals
-            # Duplicate weights for symmetric samples (m1 <-> m2)
-            weights = np.tile(weights_over_VT, 2)
+            weights = weights_over_VT
         else:
-            weights = np.tile(poisson_weights, 2)
+            weights = poisson_weights
     else:
         weights = None
 
-    # If per-point bandwidth exists use it directly, otherwise
-    # use the adaptive KDE algorithm
-    if 'perpoint_bws' in group:
+    if opts.integrate_kde == 'marginalized':
+        # ========== MARGINALIZED KDE METHOD ==========
         per_point_bandwidth = group['perpoint_bws'][...]
+        # Common parameters for all KDE calls
+        common_params = {
+            'per_point_bw': per_point_bandwidth,
+            'input_transf': ('log', 'log', 'none'),
+            'rescale_factors': [1/bwx, 1/bwy, 1/bwz],
+            'weights': weights,
+            'm1grid': m1grid,
+            'm2grid': m2grid,
+            'cfgrid': cfgrid,
+            'alpha': alpha,
+            'group': group,
+            'apply_symmetry': True,
+            'dimension_names': ['m1', 'm2', 'chieff']
+        }
+
+        # Helper function to determine constraint based on keep_dims
```

Comment:
```suggestion
        # Helper function
```
the docstring already explains what it is doing